EventListener/PsrResponseListener.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\EventListener;
+
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory;
+use Symfony\Bridge\PsrHttpMessage\HttpFoundationFactoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Converts PSR-7 Response to HttpFoundation Response using the bridge.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PsrResponseListener implements EventSubscriberInterface
+{
+    /**
+     * @var HttpFoundationFactoryInterface
+     */
+    private $httpFoundationFactory;
+
+    public function __construct(HttpFoundationFactoryInterface $httpFoundationFactory = null)
+    {
+        if (null === $httpFoundationFactory) {
+            if (!class_exists('Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory')) {
+                throw new \RuntimeException('Install the PSR-7 bridge to return a Psr\Http\Message\ResponseInterface instance in the controller.');
+            }
```

Comment:
Anyway, we should leave the message here too in case someone uses the class on their own.

What I am more concerned about is the idea to make the argument optional. Is that really a good idea? Why not injecting an `HttpFoundationFactory` instance by default? People wanting to use a different implementation will have to inject it on their own anyway.